aziotctl/aziotctl-common/src/status.rs

```diff
@@ -0,0 +1,126 @@
+use std::fmt;
+use std::process::Command;
+
+use crate::ServiceDefinition;
+
+#[allow(clippy::module_name_repetitions)]
+pub fn get_status(processes: &[&ServiceDefinition]) {
+    let results: Vec<ServiceStatus<'_>> = processes
+        .iter()
+        .map(|process| ServiceStatus {
+            service_name: process.service,
+            service_status: read_status(process.service),
+            sockets: process
+                .sockets
+                .iter()
+                .map(|socket| SocketStatus {
+                    socket_name: socket,
+                    socket_status: read_status(socket),
+                })
+                .collect(),
+        })
+        .collect();
+
+    if results.iter().any(|s| !s.ok()) {
+        for result in &results {
+            println!("{}: {}", result.service_name, result.service_status);
+            for socket in &result.sockets {
+                println!("{}: {}", socket.socket_name, socket.socket_status);
+            }
+            println!();
+        }
+
+        for result in &results {
+            if result.service_status == Status::Failed {
+                print_logs(result.service_name);
+            }
+            for socket in &result.sockets {
+                if !socket.ok() {
+                    print_logs(socket.socket_name);
+                }
+            }
+        }
+
+        println!("\n\nFor more detailed logs, use the `system logs` command. If the logs do not contain enough information, consider setting debug logs using `system set-log-level`.");
```

Comment:
nit: I'd swap out the `\n`s for calls to `println!()`.
Super inconsequential change, but IMO it makes it easier to grok what the output will look like.